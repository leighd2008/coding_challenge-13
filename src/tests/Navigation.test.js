import { shallow } from 'enzyme';
import React from 'react';
import Navigation from '../components/Navigation/Navigation';

it('expect to render Navigation component', () => {
	expect(shallow(<Navigation />)).toMatchSnapshot();
})