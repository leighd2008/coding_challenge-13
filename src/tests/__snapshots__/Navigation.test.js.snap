// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Navigation component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row center text-center"
      >
        <div
          className="d-inline-block"
        >
          <button
            className="button-primary button--icon"
            onClick={[Function]}
          >
            <i
              className="fas fa-globe"
            />
            <span
              className="d-none d-sm-block sButton"
            >
              Web
            </span>
          </button>
        </div>
        <div
          className="d-inline-block"
        >
          <button
            className="button-primary button--icon"
            onClick={[Function]}
          >
            <i
              className="fas fa-mobile-alt"
            />
            <span
              className="d-none d-sm-block sButton"
            >
              Mobile
            </span>
          </button>
        </div>
        <div
          className="d-inline-block"
        >
          <button
            className="button-primary button--icon"
            onClick={[Function]}
          >
            <i
              className="fas fa-code"
            />
            <span
              className="d-none d-sm-block sButton"
            >
              Languages
            </span>
          </button>
        </div>
        <div
          className="d-inline-block"
        >
          <button
            className="button-primary button--icon"
            onClick={[Function]}
          >
            <i
              className="fas fa-server"
            />
            <span
              className="d-none d-sm-block sButton"
            >
              Backend
            </span>
          </button>
        </div>
      </div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="d-inline-block"
          >
            <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-globe"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Web
              </span>
            </button>
          </div>,
          <div
            className="d-inline-block"
          >
            <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-mobile-alt"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Mobile
              </span>
            </button>
          </div>,
          <div
            className="d-inline-block"
          >
            <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-code"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Languages
              </span>
            </button>
          </div>,
          <div
            className="d-inline-block"
          >
            <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-server"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Backend
              </span>
            </button>
          </div>,
        ],
        "className": "row center text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-globe"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Web
              </span>
            </button>,
            "className": "d-inline-block",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-globe"
                />,
                <span
                  className="d-none d-sm-block sButton"
                >
                  Web
                </span>,
              ],
              "className": "button-primary button--icon",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-globe",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Web",
                  "className": "d-none d-sm-block sButton",
                },
                "ref": null,
                "rendered": "Web",
                "type": "span",
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-mobile-alt"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Mobile
              </span>
            </button>,
            "className": "d-inline-block",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-mobile-alt"
                />,
                <span
                  className="d-none d-sm-block sButton"
                >
                  Mobile
                </span>,
              ],
              "className": "button-primary button--icon",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-mobile-alt",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Mobile",
                  "className": "d-none d-sm-block sButton",
                },
                "ref": null,
                "rendered": "Mobile",
                "type": "span",
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-code"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Languages
              </span>
            </button>,
            "className": "d-inline-block",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-code"
                />,
                <span
                  className="d-none d-sm-block sButton"
                >
                  Languages
                </span>,
              ],
              "className": "button-primary button--icon",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-code",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Languages",
                  "className": "d-none d-sm-block sButton",
                },
                "ref": null,
                "rendered": "Languages",
                "type": "span",
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-server"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Backend
              </span>
            </button>,
            "className": "d-inline-block",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-server"
                />,
                <span
                  className="d-none d-sm-block sButton"
                >
                  Backend
                </span>,
              ],
              "className": "button-primary button--icon",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-server",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Backend",
                  "className": "d-none d-sm-block sButton",
                },
                "ref": null,
                "rendered": "Backend",
                "type": "span",
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row center text-center"
        >
          <div
            className="d-inline-block"
          >
            <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-globe"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Web
              </span>
            </button>
          </div>
          <div
            className="d-inline-block"
          >
            <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-mobile-alt"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Mobile
              </span>
            </button>
          </div>
          <div
            className="d-inline-block"
          >
            <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-code"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Languages
              </span>
            </button>
          </div>
          <div
            className="d-inline-block"
          >
            <button
              className="button-primary button--icon"
              onClick={[Function]}
            >
              <i
                className="fas fa-server"
              />
              <span
                className="d-none d-sm-block sButton"
              >
                Backend
              </span>
            </button>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="d-inline-block"
            >
              <button
                className="button-primary button--icon"
                onClick={[Function]}
              >
                <i
                  className="fas fa-globe"
                />
                <span
                  className="d-none d-sm-block sButton"
                >
                  Web
                </span>
              </button>
            </div>,
            <div
              className="d-inline-block"
            >
              <button
                className="button-primary button--icon"
                onClick={[Function]}
              >
                <i
                  className="fas fa-mobile-alt"
                />
                <span
                  className="d-none d-sm-block sButton"
                >
                  Mobile
                </span>
              </button>
            </div>,
            <div
              className="d-inline-block"
            >
              <button
                className="button-primary button--icon"
                onClick={[Function]}
              >
                <i
                  className="fas fa-code"
                />
                <span
                  className="d-none d-sm-block sButton"
                >
                  Languages
                </span>
              </button>
            </div>,
            <div
              className="d-inline-block"
            >
              <button
                className="button-primary button--icon"
                onClick={[Function]}
              >
                <i
                  className="fas fa-server"
                />
                <span
                  className="d-none d-sm-block sButton"
                >
                  Backend
                </span>
              </button>
            </div>,
          ],
          "className": "row center text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="button-primary button--icon"
                onClick={[Function]}
              >
                <i
                  className="fas fa-globe"
                />
                <span
                  className="d-none d-sm-block sButton"
                >
                  Web
                </span>
              </button>,
              "className": "d-inline-block",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-globe"
                  />,
                  <span
                    className="d-none d-sm-block sButton"
                  >
                    Web
                  </span>,
                ],
                "className": "button-primary button--icon",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-globe",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Web",
                    "className": "d-none d-sm-block sButton",
                  },
                  "ref": null,
                  "rendered": "Web",
                  "type": "span",
                },
              ],
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="button-primary button--icon"
                onClick={[Function]}
              >
                <i
                  className="fas fa-mobile-alt"
                />
                <span
                  className="d-none d-sm-block sButton"
                >
                  Mobile
                </span>
              </button>,
              "className": "d-inline-block",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-mobile-alt"
                  />,
                  <span
                    className="d-none d-sm-block sButton"
                  >
                    Mobile
                  </span>,
                ],
                "className": "button-primary button--icon",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-mobile-alt",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Mobile",
                    "className": "d-none d-sm-block sButton",
                  },
                  "ref": null,
                  "rendered": "Mobile",
                  "type": "span",
                },
              ],
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="button-primary button--icon"
                onClick={[Function]}
              >
                <i
                  className="fas fa-code"
                />
                <span
                  className="d-none d-sm-block sButton"
                >
                  Languages
                </span>
              </button>,
              "className": "d-inline-block",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-code"
                  />,
                  <span
                    className="d-none d-sm-block sButton"
                  >
                    Languages
                  </span>,
                ],
                "className": "button-primary button--icon",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-code",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Languages",
                    "className": "d-none d-sm-block sButton",
                  },
                  "ref": null,
                  "rendered": "Languages",
                  "type": "span",
                },
              ],
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="button-primary button--icon"
                onClick={[Function]}
              >
                <i
                  className="fas fa-server"
                />
                <span
                  className="d-none d-sm-block sButton"
                >
                  Backend
                </span>
              </button>,
              "className": "d-inline-block",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-server"
                  />,
                  <span
                    className="d-none d-sm-block sButton"
                  >
                    Backend
                  </span>,
                ],
                "className": "button-primary button--icon",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-server",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Backend",
                    "className": "d-none d-sm-block sButton",
                  },
                  "ref": null,
                  "rendered": "Backend",
                  "type": "span",
                },
              ],
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
